{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: matplotlib in ./.venv/lib/python3.10/site-packages (3.9.0)\n",
      "Requirement already satisfied: contourpy>=1.0.1 in ./.venv/lib/python3.10/site-packages (from matplotlib) (1.2.1)\n",
      "Requirement already satisfied: cycler>=0.10 in ./.venv/lib/python3.10/site-packages (from matplotlib) (0.12.1)\n",
      "Requirement already satisfied: fonttools>=4.22.0 in ./.venv/lib/python3.10/site-packages (from matplotlib) (4.53.0)\n",
      "Requirement already satisfied: kiwisolver>=1.3.1 in ./.venv/lib/python3.10/site-packages (from matplotlib) (1.4.5)\n",
      "Requirement already satisfied: numpy>=1.23 in ./.venv/lib/python3.10/site-packages (from matplotlib) (2.0.0)\n",
      "Requirement already satisfied: packaging>=20.0 in ./.venv/lib/python3.10/site-packages (from matplotlib) (24.1)\n",
      "Requirement already satisfied: pillow>=8 in ./.venv/lib/python3.10/site-packages (from matplotlib) (10.3.0)\n",
      "Requirement already satisfied: pyparsing>=2.3.1 in ./.venv/lib/python3.10/site-packages (from matplotlib) (3.1.2)\n",
      "Requirement already satisfied: python-dateutil>=2.7 in ./.venv/lib/python3.10/site-packages (from matplotlib) (2.9.0.post0)\n",
      "Requirement already satisfied: six>=1.5 in ./.venv/lib/python3.10/site-packages (from python-dateutil>=2.7->matplotlib) (1.16.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install matplotlib"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pandas\n",
      "  Using cached pandas-2.2.2-cp310-cp310-macosx_11_0_arm64.whl.metadata (19 kB)\n",
      "Requirement already satisfied: numpy>=1.22.4 in ./.venv/lib/python3.10/site-packages (from pandas) (2.0.0)\n",
      "Requirement already satisfied: python-dateutil>=2.8.2 in ./.venv/lib/python3.10/site-packages (from pandas) (2.9.0.post0)\n",
      "Collecting pytz>=2020.1 (from pandas)\n",
      "  Using cached pytz-2024.1-py2.py3-none-any.whl.metadata (22 kB)\n",
      "Collecting tzdata>=2022.7 (from pandas)\n",
      "  Using cached tzdata-2024.1-py2.py3-none-any.whl.metadata (1.4 kB)\n",
      "Requirement already satisfied: six>=1.5 in ./.venv/lib/python3.10/site-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)\n",
      "Using cached pandas-2.2.2-cp310-cp310-macosx_11_0_arm64.whl (11.3 MB)\n",
      "Using cached pytz-2024.1-py2.py3-none-any.whl (505 kB)\n",
      "Using cached tzdata-2024.1-py2.py3-none-any.whl (345 kB)\n",
      "Installing collected packages: pytz, tzdata, pandas\n",
      "Successfully installed pandas-2.2.2 pytz-2024.1 tzdata-2024.1\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install pandas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting pygraphviz\n",
      "  Using cached pygraphviz-1.13-cp310-cp310-macosx_10_9_universal2.whl\n",
      "Installing collected packages: pygraphviz\n",
      "Successfully installed pygraphviz-1.13\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install pygraphviz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "vscode": {
     "languageId": "shellscript"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: pyvis in ./.venv/lib/python3.10/site-packages (0.3.2)\n",
      "Requirement already satisfied: ipython>=5.3.0 in ./.venv/lib/python3.10/site-packages (from pyvis) (8.26.0)\n",
      "Requirement already satisfied: jinja2>=2.9.6 in ./.venv/lib/python3.10/site-packages (from pyvis) (3.1.4)\n",
      "Requirement already satisfied: jsonpickle>=1.4.1 in ./.venv/lib/python3.10/site-packages (from pyvis) (3.2.2)\n",
      "Requirement already satisfied: networkx>=1.11 in ./.venv/lib/python3.10/site-packages (from pyvis) (3.3)\n",
      "Requirement already satisfied: decorator in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (5.1.1)\n",
      "Requirement already satisfied: jedi>=0.16 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (0.19.1)\n",
      "Requirement already satisfied: matplotlib-inline in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (0.1.7)\n",
      "Requirement already satisfied: prompt-toolkit<3.1.0,>=3.0.41 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (3.0.47)\n",
      "Requirement already satisfied: pygments>=2.4.0 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (2.18.0)\n",
      "Requirement already satisfied: stack-data in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (0.6.3)\n",
      "Requirement already satisfied: traitlets>=5.13.0 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (5.14.3)\n",
      "Requirement already satisfied: exceptiongroup in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (1.2.1)\n",
      "Requirement already satisfied: typing-extensions>=4.6 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (4.12.2)\n",
      "Requirement already satisfied: pexpect>4.3 in ./.venv/lib/python3.10/site-packages (from ipython>=5.3.0->pyvis) (4.9.0)\n",
      "Requirement already satisfied: MarkupSafe>=2.0 in ./.venv/lib/python3.10/site-packages (from jinja2>=2.9.6->pyvis) (2.1.5)\n",
      "Requirement already satisfied: parso<0.9.0,>=0.8.3 in ./.venv/lib/python3.10/site-packages (from jedi>=0.16->ipython>=5.3.0->pyvis) (0.8.4)\n",
      "Requirement already satisfied: ptyprocess>=0.5 in ./.venv/lib/python3.10/site-packages (from pexpect>4.3->ipython>=5.3.0->pyvis) (0.7.0)\n",
      "Requirement already satisfied: wcwidth in ./.venv/lib/python3.10/site-packages (from prompt-toolkit<3.1.0,>=3.0.41->ipython>=5.3.0->pyvis) (0.2.13)\n",
      "Requirement already satisfied: executing>=1.2.0 in ./.venv/lib/python3.10/site-packages (from stack-data->ipython>=5.3.0->pyvis) (2.0.1)\n",
      "Requirement already satisfied: asttokens>=2.1.0 in ./.venv/lib/python3.10/site-packages (from stack-data->ipython>=5.3.0->pyvis) (2.4.1)\n",
      "Requirement already satisfied: pure-eval in ./.venv/lib/python3.10/site-packages (from stack-data->ipython>=5.3.0->pyvis) (0.2.2)\n",
      "Requirement already satisfied: six>=1.12.0 in ./.venv/lib/python3.10/site-packages (from asttokens>=2.1.0->stack-data->ipython>=5.3.0->pyvis) (1.16.0)\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip install pyvis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: When  cdn_resources is 'local' jupyter notebook has issues displaying graphics on chrome/safari. Use cdn_resources='in_line' or cdn_resources='remote' if you have issues viewing graphics in a notebook.\n",
      "ex.html\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "\n",
       "        <iframe\n",
       "            width=\"1500\"\n",
       "            height=\"800\"\n",
       "            src=\"ex.html\"\n",
       "            frameborder=\"0\"\n",
       "            allowfullscreen\n",
       "            \n",
       "        ></iframe>\n",
       "        "
      ],
      "text/plain": [
       "<IPython.lib.display.IFrame at 0x1157ab550>"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import networkx as nx\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "from collections import defaultdict\n",
    "from ast import literal_eval\n",
    "import textwrap\n",
    "import graphviz\n",
    "import pygraphviz\n",
    "from pyvis.network import Network\n",
    "# Read data from TSV file\n",
    "df = pd.read_csv(\"biorxiv_cancer_posts (2).tsv\", delimiter=\"\\t\")\n",
    "\n",
    "df_1 = df.drop_duplicates(subset=[\"Title\"])\n",
    "df_1 = df_1.sort_values('Collection Period', ascending = False)\n",
    "df_sampled = df_1.iloc[:25,:]\n",
    "#df_sampled = df_1.sample(n=20, random_state=42)\n",
    "\n",
    "G = nx.Graph()\n",
    "attributes = defaultdict(list)\n",
    "\n",
    "for index, row in df_sampled.iterrows():\n",
    "    title = row['Title']\n",
    "    keywords = str(row['Keywords']).split(\" \")\n",
    "    for keyword in keywords:\n",
    "        attributes[keyword].append(title)\n",
    "\n",
    "attributes = {key: list(set(value)) for key, value in attributes.items()}\n",
    "attributes = {key: value for key, value in attributes.items() if len(value) >= 2}\n",
    "\n",
    "for keyword in attributes.keys():\n",
    "    G.add_node(keyword, type='keyword')\n",
    "\n",
    "for title in set(element for sublist in attributes.values() for element in sublist):\n",
    "    wrapped_title = textwrap.fill(title, width=20)  # Adjust width as needed\n",
    "    G.add_node(wrapped_title, type='title')\n",
    "\n",
    "for keyword, related_titles in attributes.items():\n",
    "    for title in related_titles:\n",
    "        wrapped_title = textwrap.fill(title, width=20)  # Adjust width as needed\n",
    "        G.add_edge(wrapped_title, keyword)\n",
    "\n",
    "# Define node colors and sizes based on type\n",
    "node_color = {'keyword': 'lightgreen', 'title': 'skyblue'}\n",
    "node_size = {'keyword': 10000, 'title': 15000}\n",
    "\n",
    "# Assign node attributes for color and size\n",
    "for node, data in G.nodes(data=True):\n",
    "    node_type = data['type']\n",
    "    G.nodes[node]['color'] = node_color[node_type]\n",
    "    G.nodes[node]['size'] = node_size[node_type]\n",
    "\n",
    "# Draw the graph\n",
    "pos = nx.nx_agraph.graphviz_layout(G, \"sfdp\")\n",
    "#plt.figure(figsize=(40, 40))\n",
    "\n",
    "# Draw nodes\n",
    "#for node, data in G.nodes(data=True):\n",
    "#    nx.draw_networkx_nodes(G, pos, nodelist=[node], node_color=data['color'], node_size=data['size'])\n",
    "\n",
    "# Draw edges\n",
    "#nx.draw_networkx_edges(G, pos, edge_color='gray')\n",
    "\n",
    "# Draw labels\n",
    "#nx.draw_networkx_labels(G, pos, font_size=12, font_weight='bold')\n",
    "\n",
    "#plt.title(\"Graph of Paper Titles and Common Keywords\")\n",
    "#plt.axis('off')  # Turn off axis\n",
    "#plt.show()\n",
    "node_size = {'keyword': 10, 'title': 15}\n",
    "\n",
    "# Assign node attributes for color and size\n",
    "for node, data in G.nodes(data=True):\n",
    "    node_type = data['type']\n",
    "    G.nodes[node]['color'] = node_color[node_type]\n",
    "    G.nodes[node]['size'] = node_size[node_type]\n",
    "\n",
    "g = Network(height = 800, width = 1500, notebook = True)\n",
    "\n",
    "g.barnes_hut()\n",
    "for node, data in G.nodes(data=True):\n",
    "    if data['type'] == 'title':\n",
    "        g.add_node(node, size=60, color='skyblue', font={'size': 40})\n",
    "    elif data['type'] == 'keyword':\n",
    "        g.add_node(node, size=80, color='lightgreen', font={'size': 100})\n",
    "for edge in G.edges():\n",
    "    g.add_edge(edge[0], edge[1], width=12)\n",
    "#g.from_nx(G)\n",
    "g.show(\"ex.html\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "ename": "ModuleNotFoundError",
     "evalue": "No module named 'scipy'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mModuleNotFoundError\u001b[0m                       Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[50], line 57\u001b[0m\n\u001b[1;32m     54\u001b[0m         G\u001b[38;5;241m.\u001b[39madd_edge(dataset, attribute)\n\u001b[1;32m     56\u001b[0m \u001b[38;5;66;03m# Draw the graph\u001b[39;00m\n\u001b[0;32m---> 57\u001b[0m pos \u001b[38;5;241m=\u001b[39m \u001b[43mnx\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mkamada_kawai_layout\u001b[49m\u001b[43m(\u001b[49m\u001b[43mG\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m     58\u001b[0m plt\u001b[38;5;241m.\u001b[39mfigure(figsize\u001b[38;5;241m=\u001b[39m(\u001b[38;5;241m24\u001b[39m, \u001b[38;5;241m24\u001b[39m))\n\u001b[1;32m     60\u001b[0m \u001b[38;5;66;03m# Node color and size based on type\u001b[39;00m\n",
      "File \u001b[0;32m~/Documents/BroadHacks2024/.venv/lib/python3.10/site-packages/networkx/drawing/layout.py:710\u001b[0m, in \u001b[0;36mkamada_kawai_layout\u001b[0;34m(G, dist, pos, weight, scale, center, dim)\u001b[0m\n\u001b[1;32m    707\u001b[0m         pos \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mdict\u001b[39m(\u001b[38;5;28mzip\u001b[39m(G, np\u001b[38;5;241m.\u001b[39mlinspace(\u001b[38;5;241m0\u001b[39m, \u001b[38;5;241m1\u001b[39m, \u001b[38;5;28mlen\u001b[39m(G))))\n\u001b[1;32m    708\u001b[0m pos_arr \u001b[38;5;241m=\u001b[39m np\u001b[38;5;241m.\u001b[39marray([pos[n] \u001b[38;5;28;01mfor\u001b[39;00m n \u001b[38;5;129;01min\u001b[39;00m G])\n\u001b[0;32m--> 710\u001b[0m pos \u001b[38;5;241m=\u001b[39m \u001b[43m_kamada_kawai_solve\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdist_mtx\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mpos_arr\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdim\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    712\u001b[0m pos \u001b[38;5;241m=\u001b[39m rescale_layout(pos, scale\u001b[38;5;241m=\u001b[39mscale) \u001b[38;5;241m+\u001b[39m center\n\u001b[1;32m    713\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mdict\u001b[39m(\u001b[38;5;28mzip\u001b[39m(G, pos))\n",
      "File \u001b[0;32m~/Documents/BroadHacks2024/.venv/lib/python3.10/site-packages/networkx/drawing/layout.py:722\u001b[0m, in \u001b[0;36m_kamada_kawai_solve\u001b[0;34m(dist_mtx, pos_arr, dim)\u001b[0m\n\u001b[1;32m    716\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m_kamada_kawai_solve\u001b[39m(dist_mtx, pos_arr, dim):\n\u001b[1;32m    717\u001b[0m     \u001b[38;5;66;03m# Anneal node locations based on the Kamada-Kawai cost-function,\u001b[39;00m\n\u001b[1;32m    718\u001b[0m     \u001b[38;5;66;03m# using the supplied matrix of preferred inter-node distances,\u001b[39;00m\n\u001b[1;32m    719\u001b[0m     \u001b[38;5;66;03m# and starting locations.\u001b[39;00m\n\u001b[1;32m    721\u001b[0m     \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mnp\u001b[39;00m\n\u001b[0;32m--> 722\u001b[0m     \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mscipy\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01msp\u001b[39;00m\n\u001b[1;32m    724\u001b[0m     meanwt \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m1e-3\u001b[39m\n\u001b[1;32m    725\u001b[0m     costargs \u001b[38;5;241m=\u001b[39m (np, \u001b[38;5;241m1\u001b[39m \u001b[38;5;241m/\u001b[39m (dist_mtx \u001b[38;5;241m+\u001b[39m np\u001b[38;5;241m.\u001b[39meye(dist_mtx\u001b[38;5;241m.\u001b[39mshape[\u001b[38;5;241m0\u001b[39m]) \u001b[38;5;241m*\u001b[39m \u001b[38;5;241m1e-3\u001b[39m), meanwt, dim)\n",
      "\u001b[0;31mModuleNotFoundError\u001b[0m: No module named 'scipy'"
     ]
    }
   ],
   "source": [
    "g.set_options(\"\"\"\n",
    "var options = {\n",
    "  \"nodes\": {\n",
    "    \"font\": {\n",
    "      \"size\": 4\n",
    "    },\n",
    "    \"scaling\": {\n",
    "      \"min\": 10,\n",
    "      \"max\": 20\n",
    "    }\n",
    "  }\n",
    "}\n",
    "\"\"\")\n",
    "\n",
    "# Initialize a new graph\n",
    "G = nx.Graph()\n",
    "\n",
    "# Define datasets and attributes\n",
    "datasets = [\n",
    "    \"1000 Genomes Project\",\n",
    "    #\"American Gut Project\",\n",
    "    #\"Jump Cell Painting\",\n",
    "    \"Cancer Cell Line Encyclopedia (CCLE)\",\n",
    "    #\"CytoImageNet\",\n",
    "    \"ENCODE Project\",\n",
    "    \"Gene Ontology Annotations\",\n",
    "    \"The Cancer Genome Atlas Program\"\n",
    "    #\"RCSB Protein Data Bank\"\n",
    "]\n",
    "\n",
    "attributes = {\n",
    "    \"Next-Generation Sequencing\": [\"1000 Genomes Project\", \"ENCODE Project\", \"The Cancer Genome Atlas Program\"],\n",
    "    \"Human Genome\": [\"1000 Genomes Project\", \"ENCODE Project\"],\n",
    "    #\"Microbiome\": [\"American Gut Project\"],\n",
    "    #\"Cell Painting Assay\": [\"Jump Cell Painting\"],\n",
    "    #\"High-Content Imaging\": [\"Jump Cell Painting\", \"CytoImageNet\"],\n",
    "    \"Cancer\": [\"Cancer Cell Line Encyclopedia (CCLE)\", \"The Cancer Genome Atlas Program\"],\n",
    "    \"EGFR\":[\"Cancer Cell Line Encyclopedia (CCLE)\", \"Gene Ontology Annotations\", \"The Cancer Genome Atlas Program\"]\n",
    "    #\"Functional Genomics\": [\"ENCODE Project\", \"Gene Ontology Annotations\"],\n",
    "    #\"Protein Structures\": [\"RCSB Protein Data Bank\"],\n",
    "    #\"Various Genes\": [\"Gene Ontology Annotations\"]\n",
    "}\n",
    "\n",
    "# Add nodes for datasets and attributes\n",
    "for dataset in datasets:\n",
    "    G.add_node(dataset, type='dataset')\n",
    "\n",
    "for attribute in attributes.keys():\n",
    "    G.add_node(attribute, type='attribute')\n",
    "\n",
    "# Add edges between datasets and attributes\n",
    "for attribute, related_datasets in attributes.items():\n",
    "    for dataset in related_datasets:\n",
    "        G.add_edge(dataset, attribute)\n",
    "\n",
    "# Draw the graph\n",
    "pos = nx.kamada_kawai_layout(G)\n",
    "plt.figure(figsize=(24, 24))\n",
    "\n",
    "# Node color and size based on type\n",
    "node_color = ['skyblue' if G.nodes[node]['type'] == 'dataset' else 'lightgreen' for node in G]\n",
    "node_size = [1800 if G.nodes[node]['type'] == 'dataset' else 1500 for node in G]\n",
    "\n",
    "nx.draw(G, pos, with_labels=True, node_color=node_color, node_size=node_size, edge_color='gray', font_size=10, font_weight='bold')\n",
    "plt.title(\"Graph of Datasets and Common Attributes\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Title</th>\n",
       "      <th>Description</th>\n",
       "      <th>Keywords</th>\n",
       "      <th>Dataset Type</th>\n",
       "      <th>Collection Period</th>\n",
       "      <th>Organism</th>\n",
       "      <th>Genes</th>\n",
       "      <th>Tissue/Cell Type</th>\n",
       "      <th>Condition</th>\n",
       "      <th>Technique</th>\n",
       "      <th>Instrument Platform</th>\n",
       "      <th>Software</th>\n",
       "      <th>Usage Restrictions</th>\n",
       "      <th>Related Datasets</th>\n",
       "      <th>Filename</th>\n",
       "      <th>User</th>\n",
       "      <th>Post ID</th>\n",
       "      <th>Likes</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Ras-dependent activation of BMAL2 regulates hy...</td>\n",
       "      <td>To identify drivers of malignancy in human pan...</td>\n",
       "      <td>cancer pancreatic metabolism bmal2 activation ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by_nc_nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Kenneth P Olive</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Therapeutic targeting of ACLY in T-ALL in vivo</td>\n",
       "      <td>T-cell Acute Lymphoblastic Leukemia (T-ALL) is...</td>\n",
       "      <td>therapeutic vivo targeting acly leukemia lymph...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-20</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Daniel Herranz</td>\n",
       "      <td>6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Novel diagnosis- and progression-related micro...</td>\n",
       "      <td>Melanoma, an aggressive skin cancer with a hig...</td>\n",
       "      <td>melanoma microrna diagnosis signatures novel c...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by_nc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Francisco Garcia-Garcia</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Disruption of P2Y2 signaling promotes breast t...</td>\n",
       "      <td>The tumor microenvironment and wound healing a...</td>\n",
       "      <td>tumor calcium breast reducing p2y2 cancer tumo...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Makenzy L Mull</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A novel Nav1.5-dependent feedback mechanism dr...</td>\n",
       "      <td>Solid tumours have abnormally high intracellul...</td>\n",
       "      <td>cancer glycolytic breast metastasis acidificat...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-06-13</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>William J Brackenbury</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Single-cell profiling reveals the intratumor h...</td>\n",
       "      <td>BackgroundCervical adenocarcinoma (ADC) is mor...</td>\n",
       "      <td>adenocarcinoma immunosuppressive cervical intr...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-21</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Liang Weng</td>\n",
       "      <td>100</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>CDK4 and CDK6 upregulation promotes DNA replic...</td>\n",
       "      <td>Genetic interactions impact both normal human ...</td>\n",
       "      <td>cancer dna cdk4 therapy lung cancer cancers tu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Collin M. Blakely</td>\n",
       "      <td>101</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>Proteomic Analysis Reveals Trilaciclib-Induced...</td>\n",
       "      <td>Trilaciclib, a CDK4/6 inhibitor, was approved ...</td>\n",
       "      <td>trilaciclib proteomic analysis senescence reve...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by_nc_nd</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Jose Luis Marin-Rubio</td>\n",
       "      <td>102</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>Assessing Long-Term Stored Tissues for Multi-O...</td>\n",
       "      <td>As research into the complexities of cancer bi...</td>\n",
       "      <td>proteogenomics tissues suitability stored asse...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_no</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Kwang Pyo Kim</td>\n",
       "      <td>103</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>Deciphering molecular mechanisms of synergisti...</td>\n",
       "      <td>In cancer treatment, the persistent challenge ...</td>\n",
       "      <td>kinase molecular inhibitor deciphering reducti...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-14</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>cc_by</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Eirini Tsirvouli</td>\n",
       "      <td>104</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>100 rows Ã— 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                Title  \\\n",
       "0   Ras-dependent activation of BMAL2 regulates hy...   \n",
       "1      Therapeutic targeting of ACLY in T-ALL in vivo   \n",
       "2   Novel diagnosis- and progression-related micro...   \n",
       "3   Disruption of P2Y2 signaling promotes breast t...   \n",
       "4   A novel Nav1.5-dependent feedback mechanism dr...   \n",
       "..                                                ...   \n",
       "95  Single-cell profiling reveals the intratumor h...   \n",
       "96  CDK4 and CDK6 upregulation promotes DNA replic...   \n",
       "97  Proteomic Analysis Reveals Trilaciclib-Induced...   \n",
       "98  Assessing Long-Term Stored Tissues for Multi-O...   \n",
       "99  Deciphering molecular mechanisms of synergisti...   \n",
       "\n",
       "                                          Description  \\\n",
       "0   To identify drivers of malignancy in human pan...   \n",
       "1   T-cell Acute Lymphoblastic Leukemia (T-ALL) is...   \n",
       "2   Melanoma, an aggressive skin cancer with a hig...   \n",
       "3   The tumor microenvironment and wound healing a...   \n",
       "4   Solid tumours have abnormally high intracellul...   \n",
       "..                                                ...   \n",
       "95  BackgroundCervical adenocarcinoma (ADC) is mor...   \n",
       "96  Genetic interactions impact both normal human ...   \n",
       "97  Trilaciclib, a CDK4/6 inhibitor, was approved ...   \n",
       "98  As research into the complexities of cancer bi...   \n",
       "99  In cancer treatment, the persistent challenge ...   \n",
       "\n",
       "                                             Keywords  Dataset Type  \\\n",
       "0   cancer pancreatic metabolism bmal2 activation ...           NaN   \n",
       "1   therapeutic vivo targeting acly leukemia lymph...           NaN   \n",
       "2   melanoma microrna diagnosis signatures novel c...           NaN   \n",
       "3   tumor calcium breast reducing p2y2 cancer tumo...           NaN   \n",
       "4   cancer glycolytic breast metastasis acidificat...           NaN   \n",
       "..                                                ...           ...   \n",
       "95  adenocarcinoma immunosuppressive cervical intr...           NaN   \n",
       "96  cancer dna cdk4 therapy lung cancer cancers tu...           NaN   \n",
       "97  trilaciclib proteomic analysis senescence reve...           NaN   \n",
       "98  proteogenomics tissues suitability stored asse...           NaN   \n",
       "99  kinase molecular inhibitor deciphering reducti...           NaN   \n",
       "\n",
       "   Collection Period  Organism  Genes  Tissue/Cell Type  Condition  Technique  \\\n",
       "0         2024-06-22       NaN    NaN               NaN        NaN        NaN   \n",
       "1         2024-06-20       NaN    NaN               NaN        NaN        NaN   \n",
       "2         2024-06-21       NaN    NaN               NaN        NaN        NaN   \n",
       "3         2024-06-13       NaN    NaN               NaN        NaN        NaN   \n",
       "4         2024-06-13       NaN    NaN               NaN        NaN        NaN   \n",
       "..               ...       ...    ...               ...        ...        ...   \n",
       "95        2024-03-21       NaN    NaN               NaN        NaN        NaN   \n",
       "96        2024-03-14       NaN    NaN               NaN        NaN        NaN   \n",
       "97        2024-03-14       NaN    NaN               NaN        NaN        NaN   \n",
       "98        2024-03-14       NaN    NaN               NaN        NaN        NaN   \n",
       "99        2024-03-14       NaN    NaN               NaN        NaN        NaN   \n",
       "\n",
       "    Instrument Platform  Software Usage Restrictions  Related Datasets  \\\n",
       "0                   NaN       NaN        cc_by_nc_nd               NaN   \n",
       "1                   NaN       NaN              cc_no               NaN   \n",
       "2                   NaN       NaN           cc_by_nc               NaN   \n",
       "3                   NaN       NaN              cc_no               NaN   \n",
       "4                   NaN       NaN              cc_by               NaN   \n",
       "..                  ...       ...                ...               ...   \n",
       "95                  NaN       NaN              cc_by               NaN   \n",
       "96                  NaN       NaN              cc_no               NaN   \n",
       "97                  NaN       NaN        cc_by_nc_nd               NaN   \n",
       "98                  NaN       NaN              cc_no               NaN   \n",
       "99                  NaN       NaN              cc_by               NaN   \n",
       "\n",
       "    Filename                     User  Post ID  Likes  \n",
       "0        NaN          Kenneth P Olive        5      0  \n",
       "1        NaN           Daniel Herranz        6      0  \n",
       "2        NaN  Francisco Garcia-Garcia        7      0  \n",
       "3        NaN           Makenzy L Mull        8      0  \n",
       "4        NaN    William J Brackenbury        9      0  \n",
       "..       ...                      ...      ...    ...  \n",
       "95       NaN               Liang Weng      100      0  \n",
       "96       NaN        Collin M. Blakely      101      0  \n",
       "97       NaN    Jose Luis Marin-Rubio      102      0  \n",
       "98       NaN            Kwang Pyo Kim      103      0  \n",
       "99       NaN         Eirini Tsirvouli      104      0  \n",
       "\n",
       "[100 rows x 18 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
